IllegalArgumentException when using goal released-version with maven 3.6.2

When building a simple pom with goal released-version on maven 3.6.2, then the plugin crashes with an exception

**Plugin declaration:**
```
<plugin>
  <groupId>org.codehaus.mojo</groupId>
  <artifactId>build-helper-maven-plugin</artifactId>
  <version>3.0.0</version>
  <executions>
    <execution>
      <id>released-version</id>
      <goals>
        <goal>released-version</goal>
      </goals>
    </execution>
  </executions>
</plugin>
```

**Exception:**
```
Caused by: java.lang.IllegalArgumentException: version can neither be null, empty nor blank
    at org.apache.commons.lang3.Validate.notBlank (Validate.java:454)
    at org.apache.maven.artifact.ArtifactUtils.notBlank (ArtifactUtils.java:107)
    at org.apache.maven.artifact.ArtifactUtils.toSnapshotVersion (ArtifactUtils.java:57)
    at org.apache.maven.artifact.DefaultArtifact.setBaseVersionInternal (DefaultArtifact.java:389)
    at org.apache.maven.artifact.DefaultArtifact.selectVersion (DefaultArtifact.java:506)
    at org.apache.maven.artifact.DefaultArtifact.selectVersionFromNewRangeIfAvailable (DefaultArtifact.java:494)
    at org.apache.maven.artifact.DefaultArtifact.<init> (DefaultArtifact.java:106)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createArtifact (DefaultArtifactFactory.java:162)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createArtifact (DefaultArtifactFactory.java:122)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createArtifact (DefaultArtifactFactory.java:116)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createArtifact (DefaultArtifactFactory.java:44)
    at org.codehaus.mojo.buildhelper.ReleasedVersionMojo.execute (ReleasedVersionMojo.java:95)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
   ...
```

--- a/src/it/released-version/invoker.properties
+++ b/src/it/released-version/invoker.properties
@@ -1,0 +1,2 @@
+invoker.goals = test
+invoker.buildResult = success

--- a/src/it/released-version/pom.xml
+++ b/src/it/released-version/pom.xml
@@ -1,0 +1,64 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.codehaus.mojo</groupId>
+  <artifactId>build-helper-maven-plugin-released-version-it</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>build-helper-maven-plugin-released-version-it</name>
+
+  <build>
+    <defaultGoal>package</defaultGoal>
+    <plugins>
+      <plugin>
+        <groupId>@project.groupId@</groupId>
+        <artifactId>@project.artifactId@</artifactId>
+        <version>@project.version@</version>
+        <executions>
+          <execution>
+            <id>released-version</id>
+            <goals>
+              <goal>released-version</goal>
+            </goals>
+            <configuration>
+              <propertyPrefix>myReleasedVersion</propertyPrefix>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>mk-target-dir</id>
+            <phase>compile</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <tasks>
+                <mkdir dir="${project.build.directory}" />
+              </tasks>
+            </configuration>
+          </execution>
+          
+          <execution>
+            <id>echo-released-version</id>
+            <phase>test</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <tasks>
+                <echo>myReleasedVersion.version=${myReleasedVersion.version}</echo>
+                <echo>myReleasedVersion.majorVersion=${myReleasedVersion.majorVersion}</echo>
+                <echo>myReleasedVersion.minorVersion=${myReleasedVersion.minorVersion}</echo>
+                <echo>myReleasedVersion.incrementalVersion=${myReleasedVersion.incrementalVersion}</echo>
+              </tasks>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
+</project>

--- a/src/it/released-version/verify.groovy
+++ b/src/it/released-version/verify.groovy
@@ -1,0 +1,9 @@
+File file = new File( basedir, "build.log" );
+assert file.exists();
+
+String text = file.getText("utf-8");
+
+// its difficult to check the positove case with an existing released version in an IT - so currently we test only the case when no version is detected
+assert text.contains("No released version found.")
+
+return true;

--- a/src/main/java/org/codehaus/mojo/buildhelper/ReleasedVersionMojo.java
+++ b/src/main/java/org/codehaus/mojo/buildhelper/ReleasedVersionMojo.java
@@ -92,7 +92,7 @@ public class ReleasedVersionMojo
     public void execute()
     {
         org.apache.maven.artifact.Artifact artifact =
-            artifactFactory.createArtifact( getProject().getGroupId(), getProject().getArtifactId(), "", "", "" );
+            artifactFactory.createArtifact( getProject().getGroupId(), getProject().getArtifactId(), getProject().getVersion(), "", "" );
         try
         {
             ArtifactVersion releasedVersion = null;

--- a/src/main/java/org/codehaus/mojo/buildhelper/ReleasedVersionMojo.java
+++ b/src/main/java/org/codehaus/mojo/buildhelper/ReleasedVersionMojo.java
@@ -127,6 +127,9 @@ public class ReleasedVersionMojo
                 defineVersionProperty( "buildNumber", releasedVersion.getBuildNumber() );
                 defineVersionProperty( "qualifier", releasedVersion.getQualifier() );
             }
+            else {
+                getLog().debug("No released version found.");
+            }
 
         }
         catch ( ArtifactMetadataRetrievalException e )

